idaes/surrogate/tests/test_pysmo_surrogate.py

```diff
@@ -0,0 +1,2110 @@
+                        '"x_data_scaled": [[0.0, 0.0], [0.25, 0.25], [0.5, 0.5], [0.75, 0.75], [1.0, 1.0]], '
+                        '"optimal_weights": [0.02749666901085125, 0.001000000000000049], '
+                        '"optimal_p": 2, "optimal_mean": [[9.999999999902883]], "optimal_variance": [[260.13320726701056]], '
+                        '"regularization_parameter": 1e-06, '
+                        '"optimal_covariance_matrix": [[1.000001, 0.998220543300601, 0.9929011494709431, 0.9840984104422155, 0.9719055315475238], '
+                                                    '[0.998220543300601, 1.000001, 0.998220543300601, 0.9929011494709431, 0.9840984104422155], '
+                                                    '[0.9929011494709431, 0.998220543300601, 1.000001, 0.998220543300601, 0.9929011494709431], '
+                                                    '[0.9840984104422155, 0.9929011494709431, 0.998220543300601, 1.000001, 0.998220543300601], '
+                                                    '[0.9719055315475238, 0.9840984104422155, 0.9929011494709431, 0.998220543300601, 1.000001]], '
+                        '"covariance_matrix_inverse": [[108729.13455237681, -240227.09704128528, 82932.15558036882, 121970.94143487987, -73364.601633614], '
+                                                    '[-240227.0970392892, 589986.4681472526, -341158.6596781079, -130593.32427863385, 121970.94144222786], '
+                                                    '[82932.15557448889, -341158.6596663887, 516416.7835787105, -341158.659633822, 82932.15555811858], '
+                                                    '[121970.94144067129, -130593.32429416949, -341158.6596220617, 589986.4680877628, -240227.09701875152], '
+                                                    '[-73364.60163552182, 121970.94144804058, 82932.15555219717, -240227.09701673465, 108729.13454474375]], '
+                        '"optimal_y_mu": [[-3.999999999902883], [-1.999999999902883], [9.711698112369049e-11], [2.000000000097117], [4.000000000097117]], '
+                        '"training_R2": 0.9999962956250228, "training_rmse": 0.005443803800474329}, '
+                '"map": {"x_data_columns": "list", "x_data": "numpy", "x_data_min": "numpy", "x_data_max": "numpy", '
+                        '"x_data_scaled": "numpy", "optimal_weights": "numpy", "optimal_p": "str", "optimal_mean": "numpy", '
+                        '"optimal_variance": "numpy", "regularization_parameter": "str", "optimal_covariance_matrix": "numpy", '
+                        '"covariance_matrix_inverse": "numpy", "optimal_y_mu": "numpy", "training_R2": "str", "training_rmse": "str"}}}, '
+    '"input_labels": ["x1", "x2"], '
+    '"output_labels": ["z1", "z2"], '
+    '"input_bounds": {"x1": [0, 5], "x2": [0, 10]}, '
+    '"surrogate_type": "kriging"}'
+    )
+
+
+
+class TestSurrogateTrainingResult:
+
+    @pytest.fixture
+    def pysmo_outputs(self):
```

Comment:
At first glance, this really looks like it should be three separate fixtures, one for each type of trainer (and thus that each test using this should be broken into thee parts).

The problem with lumping things together like this is that if the first thing fails, you have no way of knowing if the following things would fail or not, as the test terminates at the first failure.

Reply:
Fixed.